projects/output-and-error-handling/client/main.go

```diff
@@ -0,0 +1,124 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+)
+
+const (
+	url        = "http://localhost:8080"
+	maxRetries = 3
+)
+
+func main() {
+	client := &http.Client{}
+
+	for retries := 0; retries <= maxRetries; retries++ {
+		response, err := makeWeatherRequest(url, client, retries)
+		if err != nil {
+			handleError(err)
+			os.Exit(1)
+		}
+		if response != "" {
+			fmt.Fprintln(os.Stdout, response)
+			os.Exit(0)
+		}
+	}
+}
+
+func makeWeatherRequest(url string, client *http.Client, retries int) (string, error) {
```

Comment:
Small nit: usually, we like to order parameters in terms of significance (regardless of language). Which one do you think is the most significant?

Reply:
same for `getWeather`  and `convertTime` functions